modules/bm_sim/include/bm_sim/P4Objects.h

```diff
@@ -61,7 +61,9 @@ class P4Objects {
   explicit P4Objects(std::ostream &outstream = std::cout)
       : outstream(outstream) { }
 
-  int init_objects(std::istream *is, int device_id = 0, size_t cxt_id = 0,
+  int init_objects(std::istream *is,
+                   LookupStructureFactory * lookup_factory,
```

Comment:
I do agree it would be simpler that way, the rationale for doing it this way was to leave it open to the possibility of having multiple p4 targets in the same binary.

For example, we're planning to simulate multiple switches in one binary (a network containing both RMT and NPU architectures, for example), so for that use case we would not be able to use a static factory.